Sources/ViewModels/ALKConversationViewModel.swift

```diff
@@ -1220,10 +1220,8 @@ open class ALKConversationViewModel: NSObject, Localizable {
             self.alMessages = messages.reversed() as! [ALMessage]
             self.alMessageWrapper.addObject(toMessageArray: messages)
             self.modelsToBeAddedAfterDelay = self.alMessages.map { $0.messageModel }
-            //Check for Conversation Assignee to show Typing Indicator
-            if !self.isConversationAssignedToBot() {
-                self.messageModels = self.modelsToBeAddedAfterDelay
-            } else if (UserDefaults.standard.integer(forKey: "botDelayInterval")) > 0 {
+            // Check for Conversation Assignee and conversation first message created time to show Typing Indicator.
+            if self.isConversationAssignedToBot() && (UserDefaults.standard.integer(forKey: "botDelayInterval") > 0) && !self.isOldConversation() {
```

Comment:
Why check for old conversations? Shouldn't we show it only when a new message comes after a user sends a message?

Reply:
but welcome message is been loaded from this api call in iOS(for new conversations)..mqtt notifications are after this only...